packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+
+  function test_ShouldRevertIfAlreadyInitialized() public {
+    vm.expectRevert("contract already initialized");
+    lockedGold.initialize(address(registry), unlockingPeriod);
+  }
+}
+
+contract SetRegistry is LockedGoldFoundryTest {
+  function setUp() public {
+    super.setUp();
+  }
+
+  function test_ShouldSetTheRegistryWhenCalledByTheOwner() public {
+    address newRegistry = actor("newAddress");
+    lockedGold.setRegistry(newRegistry);
+    assertEq(address(lockedGold.registry()), newRegistry);
+  }
+
+  function test_ShouldRevertWhenCalledByNonOwner() public {
+    vm.expectRevert("Ownable: caller is not the owner");
+    vm.prank(randomAddress);
+    lockedGold.setRegistry(address(0));
+  }
+}
+
+contract SetUnlockingPeriod is LockedGoldFoundryTest {
+  function setUp() public {
+    super.setUp();
+  }
+
+  function test_ShouldSetTheUnlockingPeriod() public {
+    uint256 newUnlockingPeriod = 100;
+    lockedGold.setUnlockingPeriod(newUnlockingPeriod);
+    assertEq(lockedGold.unlockingPeriod(), newUnlockingPeriod);
+  }
+
+  function test_ShouldEmitUnlockingPEriodSetEvent() public {
+    uint256 newUnlockingPeriod = 100;
+    vm.expectEmit(true, true, true, true);
+    emit UnlockingPeriodSet(newUnlockingPeriod);
+    lockedGold.setUnlockingPeriod(newUnlockingPeriod);
+  }
+
+  function test_ShouldRevertWhenUnlockingPeriodIsUnchanged() public {
+    vm.expectRevert("Unlocking period not changed");
+    lockedGold.setUnlockingPeriod(unlockingPeriod);
+  }
+
+  function test_ShouldRevertWhenCalledByNonOwner() public {
+    vm.expectRevert("Ownable: caller is not the owner");
+    vm.prank(randomAddress);
+    lockedGold.setUnlockingPeriod(100);
+  }
+}
+
+contract Lock is LockedGoldFoundryTest {
+  uint256 value = 1000;
+  function setUp() public {
+    super.setUp();
+  }
+
+  function test_ShouldIncreaseTheAccountsNonVotingLockedGoldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getAccountNonvotingLockedGold(address(this)), value);
+  }
+
+  function test_ShouldIncreaseTheAccountTOtalLockedGoldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getAccountTotalLockedGold(address(this)), value);
+  }
+
+  function test_ShouldIncreaseTheNonvotingLockedGOldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getNonvotingLockedGold(), value);
+  }
+
+  function test_ShouldIncreaseTheTotalLockedGoldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getTotalLockedGold(), value);
+  }
+
+  function test_ShouldEmitAGoldLockedEvent() public {
+    vm.expectEmit(true, true, true, true);
+    emit GoldLocked(address(this), value);
+    lockedGold.lock.value(value)();
+  }
+
+  function test_ShouldRevertWhenAccountDoesNotExist() public {
+    vm.expectRevert("Must first register address with Account.createAccount");
+    vm.prank(randomAddress);
+    lockedGold.lock();
+  }
+}
+
+contract Unlock is LockedGoldFoundryTest {
```

Comment:
The different scenarios for unlocking does not seem to all be captured here.

e.g.: It's not clear which test handles the scenario "when there are no balance requirements && when the account is not voting in governance" vs "when there are no balance requirements && when the account is voting in governance"